packages/ns-api/files/ns.threatshield

```diff
@@ -541,6 +542,141 @@ def dns_delete_bypass(e_uci, payload):
     e_uci.save('adblock')
     return {'message': 'success'}
 
+
+def __parse_expression(expression):
+    if isinstance(expression, str) or isinstance(expression, int):
+        return ipaddress.ip_address(expression)
+    if isinstance(expression, list):
+        return [__parse_expression(item) for item in expression]
+    if isinstance(expression, dict):
+        if "prefix" in expression:
+            return ipaddress.ip_network(expression["prefix"]["addr"] + "/" + str(expression["prefix"]["len"]))
+        elif "range" in expression:
+            return list(ipaddress.summarize_address_range(
+                __parse_expression(expression["range"][0]),
+                __parse_expression(expression["range"][1])
+            ))
+        elif "elem" in expression:
+            return {
+                "timeout": expression["elem"]["timeout"],
+                "expires": expression["elem"]["expires"],
+                "val": __parse_expression(expression["elem"]["val"]),
+            }
+
+    raise NotImplementedError('Unsupported expression type: {}'.format(type(expression)))
```

Comment:
Maybe returning an empty object is safer?

Reply:
It depends, if we return an empty object we lose the information that something is there, but we have no clue what it is. This way we can catch instantly the issue.

Feel like either road is fine, what do you prefer?

Reply:
As far as it's handled well by the UI, we can leave as is.

Reply:
Oh well, then it doesn't :grimacing: 

I'll return a empty object and add some logging.